### PR TITLE
fix replay button

### PR DESCRIPTION
**Pull request by elvisiraguha**
- replay button load new game instead of reloading page

[finishes #168171925]

### CODE REVIEW

File: src/gameplayGrid.js [hunk 1]
```diff
@@ -65,6 +67,12 @@ export default class GameplayGrid extends Entity {
     if (this.gameState === 'LOST') {
       canvas.writeText('You lose!', '72px sans-serif', '#FF6347', 'center', canvas.width / 2, canvas.height / 2);
       this.score.saveHighestScore(this.score.value);
+      if (this.replay.clicked) {
+        this.answerField.focus();
+        this.getNextRocket();
+        this.score.increaseValueBy(5);
+        this.replay.clicked = false;
```
Comment (Nziranziza) [inline src/gameplayGrid.js]: @elvisiraguha I think you need to define a setter within  for setting this variable(clicked)

File: src/gameplayGrid.js [hunk 2]
```diff
@@ -65,6 +67,12 @@ export default class GameplayGrid extends Entity {
     if (this.gameState === 'LOST') {
       canvas.writeText('You lose!', '72px sans-serif', '#FF6347', 'center', canvas.width / 2, canvas.height / 2);
       this.score.saveHighestScore(this.score.value);
+      if (this.replay.clicked) {
```
Comment (Nziranziza) [inline src/gameplayGrid.js]: @elvisiraguha I suggest you use getter for getting this variable(clicked)

File: src/replayButton.js [hunk 3]
```diff
@@ -1,13 +1,13 @@
-class Replay {
+export default class Replay {
```
Comment (Nziranziza) [inline src/replayButton.js]: The getter and setter I mentioned above should be inside this class

**Comment by Nziranziza** [comment]
@elvisiraguha I noticed that once I reply the game. The current score does not reset

File: src/gameplayGrid.js (hunk 1 above)
Comment (elvisiraguha) [reply src/gameplayGrid.js]: for setting it to false again??

File: src/gameplayGrid.js [hunk 4]
```diff
@@ -65,6 +67,12 @@ export default class GameplayGrid extends Entity {
...
+        this.answerField.focus();
+        this.getNextRocket();
+        this.score.increaseValueBy(5);
```
Comment (Nziranziza) [inline src/gameplayGrid.js]: you don't need to increase the score when the user replay the game. But reset score to zero

File: src/replayButton.js (hunk 3 above)
Comment (elvisiraguha) [reply src/replayButton.js]: I am going to think of a way to do it

**Comment by Nziranziza** [comment]
I also noticed that once I use clicks on the button when game is playing. The game will continue playing even if the user gives a wrong answer

**Comment by elvisiraguha** [comment]
> I also noticed that once I use clicks on the button when game is playing. The game will continue playing even if the user gives a wrong answer

I am gonna correct that too

File: src/gameplayGrid.js [hunk 5]
```diff
@@ -71,6 +73,13 @@ export default class GameplayGrid extends Entity {
       this.getNextRocket();
       this.score.increaseValueBy(5);
     }
+
+    if (this.replay.clicked) {
+      this.answerField.focus();
+      this.getNextRocket();
+      this.score.resetScore();
+      this.replay.clicked = false;
+    }
```
Comment (elvisiraguha) [inline src/gameplayGrid.js]: @Nziranziza This looks fine

**Pull request merged**